kythe/cxx/indexer/proto/testdata/basic/deprecated.proto

```diff
@@ -14,4 +20,44 @@ message M {
 
   // Fields that are annotated with [deprecated=true] should be tagged as
   // deprecated by the proto indexer.
+
+  //- @N1 defines/binding Nested1
+  //- Nested1.node/kind record
+  //- Nested1.tag/deprecated ""
+  message N1 {
+    option deprecated = true;
+
+    //- @field3 defines/binding F3
+    //- F3.tag/deprecated ""
+    optional string field3 = 1 [deprecated = true];
+  }
+
+  //- @N2 defines/binding Nested2
+  //- Nested2.node/kind record
+  //- !{ Nested2.tag/deprecated "" }
+  message N2 {
+
+    //- @field4 defines/binding F4
+    //- F4.tag/deprecated ""
+    optional string field4 = 1 [deprecated = true];
+  }
+
+
+  //- @D defines/binding EnumD
+  //- EnumD.node/kind sum
+  //- EnumD.tag/deprecated ""
+  enum D {
+    option deprecated = true;
```

Comment:
If the enum itself is deprecated, do we want to mark the individual values deprecated? In some uses of the enum, we see `VAL1`, but not `D`.

Reply:
Actually it's probably best as you have it. I don't think we propagate the parent's deprecated tag to children/fields in other languages.

Reply:
That's a really interesting idea, though—I suspect the code generator could do that, in which case the index would reflect it.

Reply:
Though the languages themselves don't necessarily propagate these things, e.g.

```java
class A {
  @Deprecated void foo() { }
}

@Deprecated
class B {
  void foo() { }
}

public class D {
  public static void main(String[] args) {
    A a = new A();
    a.foo(); // warning here
    B b = new B(); // warning here
    b.foo();  // no warning here
  }
}
```

Reply:
Right, that's why I say the code generator would have to be involved: When generating the class (say) for a protobuf enum, the generator has the option to inject the annotation not just on the class itself, but on all the appropriate members. I don't think any of the extant generators actually does that, but that's arguably the correct place to put it if someone wanted it.